### PR TITLE
OSD-17486: Isolating build system checking the validity of the rules in a new tool

### PR DESCRIPTION
**Pull request by Nikokolas3270**
The rules for all RHOBS tenants are currently all hosted in one single repository.
This repository will be split on a per tenant basis.
Aim of new tool is to avoid duplicating the CI on each repository created from the split.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,23 @@
+FROM registry.ci.openshift.org/openshift/release:golang-1.19 as builder
+
+WORKDIR /work
+
+COPY . .
+
+RUN make build
+
+FROM registry.access.redhat.com/ubi8/ubi:latest
```
[the remark this answers is not in the document]
Comment (Nikokolas3270) [reply Dockerfile:9]: Sadly no. Minimal does not come with `yum`.

File: hack/build-go-tools.sh
[the remark this answers is not in the document]
Comment (Nikokolas3270) [reply hack/build-go-tools.sh:1]: Yes, the purpose is to install `promtool`. Sadly `go install` is not working due du replace directives. I know nothing about `bingo` but I will take a look, thanks!

File: main.go [hunk 2]
```diff
@@ -0,0 +1,694 @@
...
+	}
+
+	if output, err := runAndOutputCommand("promtool", "test", "rules", testTempPath); err != nil {
+		return fmt.Errorf("failed to run 'promtool test rules'; output:\n%v", output)
+	}
+
+	return nil
+}
+
+func runTests(testsDirPath, rulesDirPath string) {
```
[the remark this answers is not in the document]
Comment (Nikokolas3270) [reply main.go:537]: I am not sure to follow you, this a method running the `promtool` unit tests. This method is no a unit test by itself.

File: .goreleaser.yaml
[the remark this answers is not in the document]
Comment (Nikokolas3270) [reply .goreleaser.yaml:1]: Yes, we could even generate `brew` packages. 😊

File: main.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (douglascamata) [reply main.go:537]: Ahhhh, I get it now. Thanks for the clarification.

**Review by douglascamata — APPROVED** [review]
LGTM and thanks for the clarifications, @Nikokolas3270 👍

File: hack/build-go-tools.sh
[the remark this answers is not in the document]
Comment (Nikokolas3270) [reply hack/build-go-tools.sh:1]: I confirm that `bingo` is a perfectly valid option.
@douglascamata: I will change my PR to use it.
Thanks for the tip.

**Pull request closed without merging**